File: java/decorator/StarbuzzCoffee/decorator/condiments/Mocha.java
package decorator.condiments;

import component.Beverage;
import decorator.CondimentDecorator;

public class Mocha extends CondimentDecorator {

    private Beverage beverage;

    public Mocha(Beverage beverage) { this.beverage = beverage; }

    @Override
    public double cost() { return beverage.cost() + 0.10; }

    @Override
    public String getDescription() { return beverage.getDescription() + ", Mocha"; }
}

File: python/decorator/StarbuzzCoffee/decorator/condiments/Mocha.py
from component.Beverage import Beverage
from decorator.CondimentDecorator import CondimentDecorator


class Mocha(CondimentDecorator):
    __beverage = Beverage()

    def __init__(self, beverage):
        super().__init__()
        self.__beverage = beverage

    def get_description(self):
        return self.__beverage.get_description() + ", Mocha"

    def cost(self):
        return self.__beverage.cost() + 2.22
